Container Clone Panic: interface conversion crash in v0.86.0 containerCreateClone function

# Bug Report: Container Clone Panic in Provider v0.86.0

**Describe the bug**
The provider crashes with a panic when attempting to clone an LXC container from a template. The error occurs in the `containerCreateClone` function with an interface conversion panic: `interface conversion: interface {} is nil, not map[string]interface {}`.

**To Reproduce**
Steps to reproduce the behavior:

1. Create a container template resource
2. Create a container clone resource that references the template
3. Run `tofu apply`
4. See panic crash in provider

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl
# Container template
module "container_template" {
  source = "./modules/container_template"

  node              = "pve-1"
  name              = "ubuntu-template"
  lxc_id            = 15000
  description       = "Ubuntu Container Template"
  tags              = ["template"]
  unprivileged      = true
  cores             = 1
  memory            = 1024
  memory_swap       = 0
  image_id          = "local:vztmpl/ubuntu-24.04-standard_24.04-2_amd64.tar.zst"
  os_type           = "ubuntu"
  disk_size         = 8
  disk_datastore    = "local-zfs"
  vnic_name         = "veth0"
  vnic_bridge       = "vmbr0"
  vlan_tag          = null
  dns_servers       = ["1.1.1.1", "8.8.8.8"]
  dns_search_domain = "local.domain"
}

# Container clone (causes crash)
module "containers" {
  source = "./modules/container_clone"

  node          = "pve-1"
  lxc_id        = 5000
  name          = "ubuntu-ct-01"
  datastore     = "local-zfs"
  template_id   = module.container_template.lxc_id
  template_node = module.container_template.node
}
```

and the output of `tofu apply`:
```text
module.containers["ubuntu_ct_01"].proxmox_virtual_environment_container.lxc: Creating...
╷
│ Error: error cloning container: Post "https://pve-1.local.domain:8006/api2/json/nodes/pve-1/lxc/5000/clone": EOF
│ 
│   with module.containers["ubuntu_ct_01"].proxmox_virtual_environment_container.lxc,
│   on modules/container_clone/main.tf line 17, in resource "proxmox_virtual_environment_container" "lxc":
│   17: resource "proxmox_virtual_environment_container" "lxc" {
│ 
╷
│ Error: Plugin did not respond
│ 
│   with module.containers["ubuntu_ct_01"].proxmox_virtual_environment_container.lxc,
│   on modules/container_clone/main.tf line 17, in resource "proxmox_virtual_environment_container" "lxc":
│   17: resource "proxmox_virtual_environment_container" "lxc" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.

Stack trace from the terraform-provider-proxmox_v0.86.0 process:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 87 [running]:
github.com/bpg/terraform-provider-proxmox/proxmox/resource_container.containerCreateClone(0x140002e6000?, {0x103934670?, 0x14000454f00?}, 0x14000281ea0)
    github.com/bpg/terraform-provider-proxmox@v0.86.0/proxmox/resource_container.go:1241 +0x11cc
github.com/bpg/terraform-provider-proxmox/proxmox/resource_container.containerCreate(0x140002e6000?, {0x103934670?, 0x14000454f00?}, 0x14000281ea0)
    github.com/bpg/terraform-provider-proxmox@v0.86.0/proxmox/resource_container.go:789 +0x78
github.com/bpg/terraform-provider-proxmox/proxmox/resource_container.containerResourceCreate({{0x103934670, 0x14000454f00}, {0x103901700?, 0x140002e6000?}}, 0x14000281ea0)
    github.com/bpg/terraform-provider-proxmox@v0.86.0/proxmox/resource_container.go:163 +0x2c
```

**Expected behavior**
The container should be successfully cloned from the template without crashing the provider. Container templates work correctly, but cloning from templates causes a panic.

**Screenshots**
N/A - This is a provider crash/panic issue.

**Additional context**
- Single or clustered Proxmox: Single node
- Proxmox version: 9.0.11
- Provider version: v0.86.0 (latest)
- Terraform/OpenTofu version: OpenTofu v1.8.5
- OS: macOS (Darwin arm64)
- Debug logs: The panic occurs specifically in `containerCreateClone` function at line 1241 in `resource_container.go`

**Technical Details:**
- Container template creation works perfectly - the issue only affects cloning operations
- The error suggests a nil interface{} is being converted to `map[string]interface{}` which fails
- This appears to be a regression as container cloning worked in previous versions
- The panic occurs in the goroutine during the `containerCreateClone` function call
- Stack trace points to line 1241 in resource_container.go where the interface conversion fails

**Workaround:**
Create containers manually through Proxmox UI

**Impact**
This completely blocks the ability to create LXC containers from templates using the clone functionality, which is a core feature for container management in Proxmox environments. Users must either:
1. Create containers manually through Proxmox UI
2. Disable container cloning functionality entirely
3. Downgrade to an older provider version (if available)

**Request**
Please prioritize this issue as it affects a fundamental feature of the provider and prevents automated container deployment workflows.

--- a/proxmoxtf/resource/container/container.go
+++ b/proxmoxtf/resource/container/container.go
@@ -1234,6 +1234,10 @@ func containerCreateClone(ctx context.Context, d *schema.ResourceData, m interfa
 		initializationIPConfig := initializationBlock[mkInitializationIPConfig].([]interface{})
 
 		for _, c := range initializationIPConfig {
+			if c == nil {
+				continue
+			}
+
 			configBlock := c.(map[string]interface{})
 			ipv4 := configBlock[mkInitializationIPConfigIPv4].([]interface{})
 

